File: go-solutions/66.go
package main

func plusOne(digits []int) []int {
	extra := 1
	for i := len(digits) - 1; i >= 0; i-- {
		remain := (digits[i] + extra) % 10
		extra = (digits[i] + extra) / 10
		digits[i] = remain
	}

	if extra != 0 {
		return append([]int{extra}, digits...)
	}

	return digits
}

File: cpp/66.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int last = 0;
        if(digits.empty()) return digits;
        digits.back()++;
        for(int i=digits.size()-1; i>=0;  i--){
            int t = digits[i] + last;
            digits[i] = t%10;
            last = t/10;
        }
        if(last){
            digits.insert(digits.begin(), last);
        }
        return digits;
    }
};